app/github_repo_fetcher.rb

```diff
@@ -62,28 +75,35 @@ def latest_commit(repo_name, path)
     }
   end
 
-  def recursively_fetch_files(repo_name, path)
-    docs = client.contents("alphagov/#{repo_name}", path:)
-    top_level_files = docs.select { |doc| doc.path.end_with?(".md") }.map do |doc|
-      data_for_github_doc(doc, repo_name)
+  def recursively_fetch_files(repo_name, path_stack)
+    repo_dir = Dir["#{REPO_DIR}#{repo_name}/#{path_stack.join('/')}/*"]
+    top_level_files = repo_dir.select { |file_path| File.file?(file_path) && file_path.end_with?(".md") }.map do |doc_path|
+      data_for_github_doc(doc_path, repo_name)
     end
-    docs.select { |doc| doc.type == "dir" }.each_with_object(top_level_files) do |dir, files|
-      files.concat(recursively_fetch_files(repo_name, dir.path))
+    repo_dir.select { |file_path| File.directory?(file_path) }.each_with_object(top_level_files) do |dir, files|
+      files.concat(recursively_fetch_files(repo_name, path_stack << File.basename(dir)))
     end
   end
 
-  def data_for_github_doc(doc, repo_name)
-    contents = HTTP.get(doc.download_url)
-    docs_path = doc.path.sub("docs/", "").match(/(.+)\..+$/)[1]
-    filename = docs_path.split("/")[-1]
-    title = ExternalDoc.title(contents) || filename
+  def data_for_github_doc(doc_path, repo_name)
```

Comment:
I think we could give this parameter a better name:

```suggestion
  def data_for_github_doc(absolute_path_to_doc, repo_name)
```